From: Christoph Hellwig <hch@lst.de>
Subject: nfs: fix nfs_swap_rw for large-folio swap
Date: Fri, 14 Jun 2024 12:03:25 +0200

The series "large folios swap-in: handle refault cases first" will permit
the MM to swap out larger than page size chunks.  NFS has all code ready
to handle this, but has a VM_BUG_ON that triggers when this happens. 
Simply remove the VM_BUG_ON to fix this use case.

Link: https://lkml.kernel.org/r/20240614100329.1203579-2-hch@lst.de
Signed-off-by: Christoph Hellwig <hch@lst.de>
Cc: Anna Schumaker <anna@kernel.org>
Cc: Steve French <sfrench@samba.org>
Cc: Trond Myklebust <trondmy@kernel.org>
Cc: Chuanhua Han <hanchuanhua@oppo.com>
Cc: Barry Song <v-songbaohua@oppo.com>
Cc: Ryan Roberts <ryan.roberts@arm.com>
Cc: Chris Li <chrisl@kernel.org>
Cc: "Huang, Ying" <ying.huang@intel.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/nfs/direct.c |    2 --
 1 file changed, 2 deletions(-)

--- a/fs/nfs/direct.c~nfs-fix-nfs_swap_rw-for-large-folio-swap
+++ a/fs/nfs/direct.c
@@ -141,8 +141,6 @@ int nfs_swap_rw(struct kiocb *iocb, stru
 {
 	ssize_t ret;
 
-	VM_BUG_ON(iov_iter_count(iter) != PAGE_SIZE);
-
 	if (iov_iter_rw(iter) == READ)
 		ret = nfs_file_direct_read(iocb, iter, true);
 	else
_
